std/experimental/perpetual/perpetual.d

```diff
@@ -0,0 +1,352 @@
+/**
+ * Map variable, POD structure or array to file
+ *   to make it shared and persistent between program invocations.
+ * Macros:
+  * Source:    $(PHOBOSSRC std/perpetual.d)
+ */
+module std.perpetual;
+//import std.stdio;
+
+private import std.file;
+private import core.stdc.stdio;
+private import std.traits;
+private import std.string;
+private import std.conv;
+private import std.algorithm;
+private import std.exception;
+version (Windows) {
+private import core.sys.windows.windows;
+private import std.utf;
+private import std.windows.syserror;
+} else version (Posix) {
+private import core.sys.posix.fcntl;
+private import core.sys.posix.unistd;
+private import core.sys.posix.sys.mman;
+private import core.sys.posix.sys.stat;
+} else {
+	static assert(0);
+}
+
+
+
+/**
+ * Persistently maps value type object to file
+ */
+class Perpetual(T)
```

Comment:
Should be struct with either ref-counting or disabled this. There is no way to duplicate mapping